dev/io.openliberty.microprofile.health.3.0.internal/src/io/openliberty/microprofile/health30/internal/HealthCheck30HttpResponseBuilder.java

```diff
@@ -66,16 +67,23 @@ public void addResponses(Set<HealthCheckResponse> hcResponseSet) {
 
     public void setHttpResponse(HttpServletResponse httpResponse) {
         httpResponse.setHeader(HealthCheckConstants.HTTP_HEADER_CONTENT_TYPE, HealthCheckConstants.MEDIA_TYPE_APPLICATION_JSON);
-        HashMap<String, Object> payload = new HashMap<String, Object>();
 
         // Set the HTTP Response code
         httpResponse.setStatus(overallStatus == Status.UP ? 200 : 503);
 
         // Populate the payload with the overall status and checks array
-        payload.put(HealthCheckConstants.HEALTH_CHECK_PAYLOAD_STATUS, overallStatus);
-        payload.put(HealthCheckConstants.HEALTH_CHECK_PAYLOAD_CHECKS, checks.toArray());
+
+        JsonArrayBuilder jsonArrayBuilder = jsonBuilderFactory.createArrayBuilder();
```

Comment:
With using the 1.1 API, can this be improved to pass the checks collection into createArrayBuilder signature that takes a Collection?

Similarly with the createObjectBuilder, can it do the payload Map like it did before and pass that into the createObjectBuilder signature that takes a Map.

Reply:
Ah, I didn't know that was an option - it sure would be cleaner.